### ISSUE TITLE
Require path resolution not working as documented (at least on Windows)

### ISSUE THREAD

**Issue by stefandd** [crystal-lang/crystal#11842]
Despite the documentation saying that `By default the require path is the location of the standard library that comes with the compiler, and the "lib" directory relative to the current working directory (given by pwd in a Unix shell).` this does not work, at least on Windows with 1.3.2

Consider the following folder structure
```Crystal
- project
  foo.cr
    - lib
      - bar.cr
```
Content of foo.cr
```Crystal
require "bar"

say_hello()
```
Content of bar.cr
```Crystal
def say_hello
  puts "Hello Penny!"
end
```
This produces the error
```
In foo.cr:1:1

 1 | require "bar"
     ^
Error: can't find file 'bar'
```

**Comment by stefandd** [comment]
@Blacksmoke16 Does your adding of the platform:windows label mean this works as expected on Unix-likes?

**Comment by Blacksmoke16** [comment]
@stefandd Yes, if I had to guess I would say that on windows the `CRYSTAL_PATH` environmental variable isn't set correctly, which is what defines where the compiler searches for required files. For example:

```sh
$ crystal env CRYSTAL_PATH
lib:/usr/lib/crystal
```

Could try running that on windows and see if its empty or what. Workaround would be to manually set that var to point to `lib` and wherever your source code is.

Ref: https://crystal-lang.org/reference/1.3/using_the_compiler/index.html#environment-variables

**Comment by Blacksmoke16** [comment]
Actually this might be fixed next release as part of https://github.com/crystal-lang/crystal/pull/11524.

**Comment by stefandd** [comment]
> @stefandd Yes, if I had to guess I would say that on windows the `CRYSTAL_PATH` environmental variable isn't set correctly, 

Just to note: I actually had correctly set the `CRYSTAL_PATH` variable by `set CRYSTAL_PATH=c:\crystal\src`

**Comment by straight-shoota** [comment]
@Blacksmoke16 No, https://github.com/crystal-lang/crystal/pull/11524 is only about the helper tool for stdlib and compiler development.

You would not use that wrapper when running the compiler normally.

@stefandd If you want the compiler to pick up libs in `lib` folder, you need to add `lib` to `CRYSTAL_PATH`.
How did you install crystal?

**Comment by stefandd** [comment]
> If you want the compiler to pick up libs in `lib` folder, you need to add `lib` to `CRYSTAL_PATH`.

But this is contrary to what the documentation says that I quoted above. It states that it will *always* look at the lib folder relative to the current folder (without any mention that enviroment variables need to be set)

**Comment by straight-shoota** [comment]
The documentation states that this is the default behaviour. If you set the environment variable `CRYSTAL_PATH` to choose a *different* path than the compiler's default, the stated behaviour no longer applies.

Usually, you should not need to specify `CRYSTAL_PATH`. This depends on how the compiler was built, though. If done correctly, it has a default path baked in that finds the location of the standard library relative to the location of the compiler binary, and also includes the `lib` folder relative to the working directory.

How did you install Crystal and why did you need to set `CRYSTAL_PATH`?

**Comment by stefandd** [comment]
> How did you install Crystal and why did you need to set `CRYSTAL_PATH`?

Indeed, if I do not define `CRYSTAL_PATH`, this works as expected: finding the relative lib folder in whichever folder the compiler is run from. Should I close this issue, even though this behavior is imo not properly documented. With the wording from [https://crystal-lang.org/reference/1.3/syntax_and_semantics/requiring_files.html](url) that I quoted above it should not at all matter whether or not  `CRYSTAL_PATH` is defined since it states that a folder 'lib' relative to the current working directory is searched.

**Comment by Blacksmoke16** [comment]
> Indeed, if I do not define `CRYSTAL_PATH`, this works as expected

Oh well, if you're mucking with that variable to not include `lib`, then yea this is working as designed. Your issue made it seem like you were just using a default install of it and running into this problem.

Your quote covers the default value of `CRYSTAL_PATH`, once you manually edit it, it's no longer using the defaults so ofc it wouldn't work w/o `lib` included in it.

**Comment by stefandd** [comment]
> Oh well, if you're mucking with that variable to not include `lib`, then yea this is working as designed. Your issue made it seem like you were just using a default install of it and running into this problem.

It may be working as designed, but not as described in the docs. The `CRYSTAL_PATH` variable presumably controls the `default the require path` but why should it alter that the compiler is supposed to in addition look in `the "lib" directory relative to the current working directory (given by pwd in a Unix shell).` ?

**Comment by straight-shoota** [comment]
I think the wording is quite unambiguous that the require path is by default the location of stdlib + `lib`. It seems like a good idea to expand on that, though and describe what happens when you set `CRYSTAL_PATH`.

**Comment by asterite** [comment]
In my mind `CRYSTAL_PATH` is an artifact that the compiler uses to be able to distribute itself and have everything set up properly. I think we should encourage users to **not** modify this value. Everything works perfectly fine if you don't modify it.

**Comment by stefandd** [comment]
@asterite  Should I simply close this then?

**Comment by stefandd** [comment]
@asterite I agree. I just checked the reference and `CRYSTAL_PATH`'s function is explained at [https://crystal-lang.org/reference/1.3/using_the_compiler/index.html](url) The problem is that the wording there does not make clear that besides revectoring the default require search path, setting this env variable also breaks the relative `lib` folder search. So maybe this could be clarified.
